### PR TITLE
add arm64osx automated tests

### PR DESCRIPTION
**Pull request by burgerrg**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
         id: build-clone-main
         run: |
           git clone --filter=blob:none --branch main https://github.com/cisco/ChezScheme.git
-          echo "HASH_KEY=$(cat ChezScheme/.git/refs/heads/main)" >> "${GITHUB_OUTPUT}"
+          echo "HASH_KEY=2$(cat ChezScheme/.git/refs/heads/main)" >> "${GITHUB_OUTPUT}"
```
Comment (payneca) [inline .github/workflows/test.yml]: Is this `2` a typo, or does it do something I don't know about. Trying it locally seems to just prepend `2` to the hash which doesn't seem correct.

**Review by payneca — COMMENTED** [review]
In an upcoming commit, we noticed that we aren't using coreutils' `ln` either. Our solution there was to explicitly update `PATH` to catch all of coreutils. Would that be a better solution to the `realpath` problem?

File: .github/workflows/test.yml (hunk 1 above)
Comment (burgerrg) [reply .github/workflows/test.yml]: It's important, because without it, the current cache won't have arm64osx in it.

**Comment by burgerrg** [comment]
Yes, updating PATH to include all of coreutils would be fine. Be sure to use $(brew --prefix) to get the proper prefix for both Intel and ARM Macs.

**Comment by burgerrg** [comment]
If you do add "$(brew --prefix)/opt/coreutils/libexec/gnubin" to the PATH, then you should update the README and configure scripts that use ginstall and grealpath.

**Comment by burgerrg** [comment]
@payneca, I removed the changes to README and configure and simply updated GITHUB_PATH in the automated tests. Feel free to update this in a future pull request when you update everything else.

File: .github/workflows/test.yml (hunk 1 above)
Comment (owaddell) [reply .github/workflows/test.yml]: It seems the `2` is essentially marking a "schema change" for the cache. What do you think about changing the `2` to something like `v2_` to help us recognize this in the future?

(When I saw a `2` before process substitution, I initially wondered if you were doing something novel with stderr.)

Comment (burgerrg) [reply .github/workflows/test.yml]: Great idea, done!

**Comment by burgerrg** [comment]
@owaddell, I also changed fail-fast to false so that each machine type gets the opportunity to complete should any of the others fail.

**Review by payneca — APPROVED** [review]

**Review by owaddell — APPROVED** [review]

**Pull request merged**